packages/stitch/src/mergeValidations.ts

```diff
@@ -0,0 +1,251 @@
+    finalInputFieldConfig,
+    inputFieldConfigs,
+    'input field',
+    inputFieldNamespace,
+    typeMergingOptions
+  );
+
+  if (
+    getValidationSettings(inputFieldNamespace, typeMergingOptions).strictNullComparison &&
+    candidates.some(c => finalInputFieldNull !== isNonNullType(c.inputFieldConfig.type))
+  ) {
+    validationMessage(
+      `Nullability of input field "${inputFieldNamespace}" does not match across subschemas. Disable typeMergingOptions.validationSettings.strictNullComparison to permit safe divergences.`,
+      inputFieldNamespace,
+      typeMergingOptions
+    );
+  } else if (!finalInputFieldNull && candidates.some(c => isNonNullType(c.inputFieldConfig.type))) {
+    validationMessage(
+      `Canonical definition of input field "${inputFieldNamespace}" permits null while some subschemas require not-null. This will be an automatic error in future versions.`,
+      inputFieldNamespace,
+      typeMergingOptions
+    );
+  }
+
+  if (isEnumType(finalInputFieldType)) {
+    validateInputEnumConsistency(finalInputFieldType, inputFieldConfigs, typeMergingOptions);
+  }
+}
+
+export function validateTypeConsistency(
+  finalElementConfig: GraphQLFieldConfig<any, any> | GraphQLArgumentConfig | GraphQLInputFieldConfig,
+  candidates: Array<GraphQLFieldConfig<any, any> | GraphQLArgumentConfig | GraphQLInputFieldConfig>,
+  definitionType: string,
+  settingNamespace: string,
+  typeMergingOptions: TypeMergingOptions
+): void {
+  const finalNamedType = getNamedType(finalElementConfig.type);
+  const finalIsScalar = isScalarType(finalNamedType);
+  const finalIsList = hasListType(finalElementConfig.type);
+
+  candidates.forEach(c => {
+    if (finalIsList !== hasListType(c.type)) {
+      throw new Error(
+        `Definitions of ${definitionType} "${settingNamespace}" implement inconsistent list types across subschemas and cannot be merged.`
+      );
+    }
+
+    const currentNamedType = getNamedType(c.type);
+
+    if (finalNamedType.toString() !== currentNamedType.toString()) {
+      const proxyableScalar = !!typeMergingOptions?.validationSettings?.proxyableScalars?.[
+        finalNamedType.toString()
+      ]?.includes(currentNamedType.toString());
+      const bothScalars = finalIsScalar && isScalarType(currentNamedType);
+      const permitScalar = proxyableScalar && bothScalars;
+      if (proxyableScalar && !bothScalars) {
+        throw new Error(`Types ${finalNamedType} and ${currentNamedType} are not proxyable scalars.`);
+      }
+      if (!permitScalar) {
+        validationMessage(
+          `Definitions of ${definitionType} "${settingNamespace}" implement inconsistent named types across subschemas. This will be an automatic error in future versions.`,
+          settingNamespace,
+          typeMergingOptions
+        );
+      }
+    }
+  });
+}
+
+function hasListType(type: GraphQLType): boolean {
+  return isNonNullType(type) ? isListType(type.ofType) : isListType(type);
```

Comment:
Can you just call isListType(getNullableType(type))

Reply:
Done in fd789f6.